src/lib/core/utils/thumbnails.tsx

```diff
@@ -0,0 +1,104 @@
+import { renderToStaticMarkup } from 'react-dom/server';
+
+import { range, zipWith } from 'lodash';
+
+import { isFacetedPlotRef } from '@veupathdb/components/lib/types/guards';
+import { FacetedPlotRef, PlotRef } from '@veupathdb/components/lib/types/plots';
+
+export interface ThumbnailDimensions {
+  width: number;
+  height: number;
+}
+
+export function makePlotThumbnailUrl(
+  plotRef: FacetedPlotRef | PlotRef,
+  thumbnailDimensions: ThumbnailDimensions
+) {
+  if (!isFacetedPlotRef(plotRef)) {
+    return plotRef.toImage({
+      format: 'svg',
+      ...thumbnailDimensions,
+    });
+  } else {
+    return makeFacetedPlotThumbnailUrl(plotRef, thumbnailDimensions);
+  }
+}
+
+async function makeFacetedPlotThumbnailUrl(
+  facetedPlotRef: FacetedPlotRef,
+  thumbnailDimensions: ThumbnailDimensions
+) {
+  const plotImageUrls = await Promise.all(
+    facetedPlotRef
+      .slice(0, 4)
+      .map((plotRef) =>
+        plotRef == null
+          ? null
+          : plotRef.toImage({ format: 'svg', ...thumbnailDimensions })
+      )
+  );
+
+  const plotImageWidth = thumbnailDimensions.width / 2;
+  const plotImageHeight = thumbnailDimensions.height / 2;
+
+  const plotImagePositions = [
+    {
+      x: 0,
+      y: 0,
+    },
+    {
+      x: plotImageWidth,
+      y: 0,
+    },
+    {
+      x: 0,
+      y: plotImageHeight,
+    },
+    {
+      x: plotImageWidth,
+      y: plotImageHeight,
+    },
+  ];
+
+  const plotImageNodes = zipWith(
+    plotImageUrls,
+    plotImagePositions,
+    range(4),
+    (plotImageUrl, plotImagePosition, plotIndex) =>
+      plotImageUrl == null ? null : (
+        <image
+          key={`facet-plot-${plotIndex}`}
+          href={plotImageUrl}
+          width={plotImageWidth}
+          height={plotImageHeight}
+          {...plotImagePosition}
+        />
+      )
+  );
+
+  const thumbnailNode = (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      width={thumbnailDimensions.width}
+      height={thumbnailDimensions.height}
+      viewBox={`0 0 ${thumbnailDimensions.width} ${thumbnailDimensions.height}`}
+    >
+      {plotImageNodes}
+      <text
+        style={{
+          fontFamily: 'sans-serif',
+        }}
+        x={thumbnailDimensions.width * 0.95}
+        y={thumbnailDimensions.height * 0.1}
+        fontSize={thumbnailDimensions.height * 0.1}
+        textAnchor="end"
```

Comment:
The number clashes with the mouse-over icons (trash can and duplicate), so I played around with a larger central number, but I'm sure we'll get more opinions from outreach!
```
        x={thumbnailDimensions.width * 0.5}
        y={thumbnailDimensions.height * 0.55}
        fontSize={thumbnailDimensions.height * 0.2}
        textAnchor="middle"
```
![image](https://user-images.githubusercontent.com/308639/141202666-5db8030e-d4f8-4073-9241-c34609aaa96b.png)
![image](https://user-images.githubusercontent.com/308639/141203094-21616dce-32e0-4c9b-8c00-f0cd99807867.png)

Reply:
What about the same size font that is current in use, but on the bottom right?

Reply:
In the short term, I went with a variation of @bobular's suggestion (centered, with a slightly smaller font size than proposed).

This implementation of @dmfalke's suggestion holds promise, but sometimes "clashes" with the lower-right-hand-plot:

```
        x={thumbnailDimensions.width * 0.925}
        y={thumbnailDimensions.height * 0.925}
        fontSize={thumbnailDimensions.height * 0.1}
        textAnchor="end"
```

![image](https://user-images.githubusercontent.com/15959217/141369063-9424b2bf-d60f-4af2-b155-502ba526b503.png)